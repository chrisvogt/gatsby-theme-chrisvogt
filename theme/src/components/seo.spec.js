import { mount } from 'enzyme'
import { useThemeUI } from 'theme-ui'
import React from 'react'
import Helmet from 'react-helmet'

import SEO from './seo'
import useSiteMetadata from '../hooks/use-site-metadata'

jest.mock('../hooks/use-site-metadata')
jest.mock('theme-ui')

const mockSiteMetadata = {
  languageCode: 'en',
  title: 'My Personal Blog & Portfolio',
  titleTemplate: '%s | My Test Site'
}

const mockBackgroundColor = '#aeaeae'
const mockThemeUIResult = {
  theme: {
    colors: {
      background: mockBackgroundColor
    }
  }
}

describe('SEO', () => {
  useSiteMetadata.mockImplementation(() => mockSiteMetadata)
  useThemeUI.mockImplementation(() => mockThemeUIResult)

  describe('page title', () => {
    it('defaults to the site title', () => {
      mount(<SEO />)
      expect(Helmet.peek().title).toBe(
        'My Personal Blog & Portfolio | My Test Site'
      )
    })

    it('accepts a custom page title', () => {
      mount(<SEO title='derp' />)
      expect(Helmet.peek().title).toBe('derp | My Test Site')
    })
  })
})
