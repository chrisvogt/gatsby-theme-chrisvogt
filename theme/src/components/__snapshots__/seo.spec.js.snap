// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Seo handles missing title template 1`] = `
[
  <title>
    Test Site
  </title>,
  <meta
    content="var(--theme-ui-colors-background)"
    name="theme-color"
  />,
  <meta
    content="Test Site"
    property="og:title"
  />,
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Test Site"
    name="twitter:title"
  />,
  <link
    href="https://webmention.io/test.com/webmention"
    rel="webmention"
  />,
]
`;

exports[`Seo handles title template replacement 1`] = `
[
  <title>
    Custom Page Title Template
  </title>,
  <meta
    content="var(--theme-ui-colors-background)"
    name="theme-color"
  />,
  <meta
    content="Custom Page Title Template"
    property="og:title"
  />,
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Custom Page Title Template"
    name="twitter:title"
  />,
  <link
    href="https://webmention.io/test.com/webmention"
    rel="webmention"
  />,
]
`;

exports[`Seo matches the snapshot with basic props 1`] = `
[
  <title>
    Test Page | Test Site
  </title>,
  <meta
    content="Test description"
    name="description"
  />,
  <meta
    content="var(--theme-ui-colors-background)"
    name="theme-color"
  />,
  <meta
    content="Test Page | Test Site"
    property="og:title"
  />,
  <meta
    content="Test description"
    property="og:description"
  />,
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Test Page | Test Site"
    name="twitter:title"
  />,
  <meta
    content="Test description"
    name="twitter:description"
  />,
  <link
    href="https://webmention.io/test.com/webmention"
    rel="webmention"
  />,
]
`;

exports[`Seo renders with all optional props 1`] = `
[
  <title>
    Test Page | Test Site
  </title>,
  <meta
    content="Test description"
    name="description"
  />,
  <meta
    content="https://example.com/image.jpg"
    name="image"
  />,
  <meta
    content="var(--theme-ui-colors-background)"
    name="theme-color"
  />,
  <meta
    content="Test Page | Test Site"
    property="og:title"
  />,
  <meta
    content="test, keywords"
    name="keywords"
  />,
  <meta
    content="article"
    property="og:type"
  />,
  <meta
    content="Test description"
    property="og:description"
  />,
  <meta
    content="https://example.com/image.jpg"
    property="og:image"
  />,
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Test Page | Test Site"
    name="twitter:title"
  />,
  <meta
    content="https://example.com/image.jpg"
    name="twitter:image"
  />,
  <meta
    content="Test description"
    name="twitter:description"
  />,
  <link
    href="https://webmention.io/test.com/webmention"
    rel="webmention"
  />,
]
`;

exports[`Seo renders with children 1`] = `
[
  <title>
    Test Page | Test Site
  </title>,
  <meta
    content="var(--theme-ui-colors-background)"
    name="theme-color"
  />,
  <meta
    content="Test Page | Test Site"
    property="og:title"
  />,
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Test Page | Test Site"
    name="twitter:title"
  />,
  <link
    href="https://webmention.io/test.com/webmention"
    rel="webmention"
  />,
  <meta
    content="value"
    name="custom"
  />,
]
`;

exports[`Seo renders without article prop 1`] = `
[
  <title>
    Test Page | Test Site
  </title>,
  <meta
    content="Test description"
    name="description"
  />,
  <meta
    content="var(--theme-ui-colors-background)"
    name="theme-color"
  />,
  <meta
    content="Test Page | Test Site"
    property="og:title"
  />,
  <meta
    content="Test description"
    property="og:description"
  />,
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Test Page | Test Site"
    name="twitter:title"
  />,
  <meta
    content="Test description"
    name="twitter:description"
  />,
  <link
    href="https://webmention.io/test.com/webmention"
    rel="webmention"
  />,
]
`;

exports[`Seo renders without description 1`] = `
[
  <title>
    Test Page | Test Site
  </title>,
  <meta
    content="var(--theme-ui-colors-background)"
    name="theme-color"
  />,
  <meta
    content="Test Page | Test Site"
    property="og:title"
  />,
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Test Page | Test Site"
    name="twitter:title"
  />,
  <link
    href="https://webmention.io/test.com/webmention"
    rel="webmention"
  />,
]
`;

exports[`Seo renders without image 1`] = `
[
  <title>
    Test Page | Test Site
  </title>,
  <meta
    content="Test description"
    name="description"
  />,
  <meta
    content="var(--theme-ui-colors-background)"
    name="theme-color"
  />,
  <meta
    content="Test Page | Test Site"
    property="og:title"
  />,
  <meta
    content="Test description"
    property="og:description"
  />,
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Test Page | Test Site"
    name="twitter:title"
  />,
  <meta
    content="Test description"
    name="twitter:description"
  />,
  <link
    href="https://webmention.io/test.com/webmention"
    rel="webmention"
  />,
]
`;

exports[`Seo renders without keywords 1`] = `
[
  <title>
    Test Page | Test Site
  </title>,
  <meta
    content="Test description"
    name="description"
  />,
  <meta
    content="var(--theme-ui-colors-background)"
    name="theme-color"
  />,
  <meta
    content="Test Page | Test Site"
    property="og:title"
  />,
  <meta
    content="Test description"
    property="og:description"
  />,
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Test Page | Test Site"
    name="twitter:title"
  />,
  <meta
    content="Test description"
    name="twitter:description"
  />,
  <link
    href="https://webmention.io/test.com/webmention"
    rel="webmention"
  />,
]
`;

exports[`Seo renders without twitter username 1`] = `
[
  <title>
    Test Page | Test Site
  </title>,
  <meta
    content="Test description"
    name="description"
  />,
  <meta
    content="var(--theme-ui-colors-background)"
    name="theme-color"
  />,
  <meta
    content="Test Page | Test Site"
    property="og:title"
  />,
  <meta
    content="Test description"
    property="og:description"
  />,
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Test Page | Test Site"
    name="twitter:title"
  />,
  <meta
    content="Test description"
    name="twitter:description"
  />,
  <link
    href="https://webmention.io/test.com/webmention"
    rel="webmention"
  />,
]
`;

exports[`Seo renders without webmention URL 1`] = `
[
  <title>
    Test Page | Test Site
  </title>,
  <meta
    content="Test description"
    name="description"
  />,
  <meta
    content="var(--theme-ui-colors-background)"
    name="theme-color"
  />,
  <meta
    content="Test Page | Test Site"
    property="og:title"
  />,
  <meta
    content="Test description"
    property="og:description"
  />,
  <meta
    content="summary_large_image"
    name="twitter:card"
  />,
  <meta
    content="Test Page | Test Site"
    name="twitter:title"
  />,
  <meta
    content="Test description"
    name="twitter:description"
  />,
]
`;
